src/connectors/queue/payTo/blockchain.ts

```diff
@@ -158,58 +167,55 @@ export class PayToByBlockchainQueue extends BaseQueue {
       }),
     ])
 
-    const creatorAddress = recipient.ethAddress
-    const curatorAddress = sender.ethAddress
-    const cid = article.dataHash
-    const tokenAddress = polygonUSDTContractAddress
-    const amount = tx.amount
-    const decimals = polygonUSDTContractDecimals
-
-    // txReceipt does not match with tx record in database
-    if (
-      !(await this.containMatchedEvent(txReceipt.events, {
-        creatorAddress,
-        curatorAddress,
-        cid,
-        tokenAddress,
-        amount,
-        decimals,
-      }))
-    ) {
-      await this.updateTxAndBlockchainTxState(
-        {
-          txId,
-          txState: TRANSACTION_STATE.canceled,
-          txRemark: TRANSACTION_REMARK.INVALID,
-        },
-        {
-          blockchainTxId: blockchainTx.id,
-          blockchainTxState: BLOCKCHAIN_TRANSACTION_STATE.succeeded,
-        },
-        paymentService
-      )
+    // cancel tx and success blockchain tx if it's invalid
+    // Note: sender and recipient's ETH address may change after tx is created
+    const isValidTx = await this.containMatchedEvent(txReceipt.events, {
+      cid: article.dataHash,
+      amount: tx.amount,
+      // support USDT only for now
+      tokenAddress: polygonUSDTContractAddress,
+      decimals: polygonUSDTContractDecimals,
+    })
+    if (!isValidTx) {
+      await this.updateTxAndBlockchainTxState({
+        txId,
+        txState: TRANSACTION_STATE.canceled,
+        txRemark: TRANSACTION_REMARK.INVALID,
+        blockchainTxId: blockchainTx.id,
+        blockchainTxState: BLOCKCHAIN_TRANSACTION_STATE.succeeded,
+      })
       return data
     }
 
-    // update pending tx
-    await this.succeedBothTxAndBlockchainTx(
-      txId,
-      blockchainTx.id,
-      paymentService
-    )
+    // anonymize tx if sender's ETH address is not matched
+    const isSenderMatched = txReceipt.events
+      .map((e) => e.curatorAddress)
+      .every((address) => ignoreCaseMatch(address, sender.ethAddress))
+    if (!isSenderMatched) {
```

Comment:
alter `tx.senderId` to null if donor's (user) address is not the same as blockchain tx sender address (from)